test/system/src/test/java/io/pravega/test/system/ControllerFailoverTest.java

```diff
@@ -179,10 +188,107 @@ public void failoverTest() throws InterruptedException, ExecutionException {
         Assert.assertEquals(2, streamSegments.getSegments().size());
     }
 
+    @Test
+    public void sweepOverTest() throws InterruptedException, ExecutionException {
+        String scope = "testSweepOverScope" + RandomStringUtils.randomAlphabetic(5);
+        String stream = "testSweepOverStream" + RandomStringUtils.randomAlphabetic(5);
+        String readerGroupName = "testSweepOverRG" + RandomStringUtils.randomAlphabetic(5);
+        int initialSegments = 1;
+        List<Long> segmentsToSeal = Collections.singletonList(0L);
+        Map<Double, Double> newRangesToCreate = new HashMap<>();
+        newRangesToCreate.put(0.0, 1.0);
+
+        ClientConfig clientConfig = Utils.buildClientConfig(controllerURIDirect);
+        // Connect with controller instance
+        final Controller controller = new ControllerImpl(
+                ControllerImplConfig.builder()
+                        .clientConfig(clientConfig)
+                        .build(), executorService);
+
+        // Create scope, stream, and a transaction with high timeout value.
+        controller.createScope(scope).join();
+        log.info("Scope {} created successfully", scope);
+
+        createStream(controller, scope, stream, ScalingPolicy.fixed(initialSegments));
+        log.info("Stream {}/{} created successfully", scope, stream);
+
+        StreamImpl stream1 = new StreamImpl(scope, stream);
+
+        // Initiate scale operation. It will block until ongoing transaction is complete.
+        controller.startScale(stream1, segmentsToSeal, newRangesToCreate);
+
+        // Initiate and Create RG instances
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, clientConfig);
+        readerGroupManager.createReaderGroup(readerGroupName,
+                ReaderGroupConfig.builder().stream(io.pravega.client.stream.Stream.of(scope, stream)).build());
+
+        int startInclusive = 1;
+        int endExclusive = 500;
+        log.info("Write events with range [{},{})", startInclusive, endExclusive);
+        writeEvents(scope, IntStream.range(startInclusive, endExclusive).boxed().collect(Collectors.toList()), stream);
+
+        // Now stop the controller instance executing scale operation.
+        Futures.getAndHandleExceptions(controllerService.scaleService(0), ExecutionException::new);
```

Comment:
Are we sure that the future returned @ line 218 did not complete when you made the controllerService instance count as 0? If the future completes then it would imply the sweepover never succeeded.

Reply:
Yes this is a valid point.

Reply:
I did add an info log to validate the same and couldn't find an issue there. Have added a few more scenarios now.